packages/waku/src/lib/hono/engine.ts

```diff
@@ -59,7 +62,13 @@ export const serverEngine = (options: MiddlewareOptions): MiddlewareHandler => {
         }
       });
     };
-    await run(0);
+    const renderStore = {
+      context: ctx.context,
+      rerender: () => {
+        throw new Error('Cannot rerender');
+      },
+    };
+    await runWithRenderStoreInternal(renderStore, () => run(0));
```

Comment:
This doesn't looks nice. This function shouldn't be used outside RSC capabilities.
Consider `unstable_getHonoContext` is for server components.
Middleware can use `__hono_context` directly.
A little concern of mine is there can be two types of middleware, one that depends on hono and the other that doesn't.